server/sonar-server/src/main/java/org/sonar/server/permission/ws/SearchProjectPermissionsAction.java

```diff
@@ -113,19 +118,15 @@ private static SearchProjectPermissionsWsRequest toSearchProjectPermissionsWsReq
       .setQuery(request.param(Param.TEXT_QUERY));
   }
 
-  private void checkRequestAndPermissions(SearchProjectPermissionsWsRequest request) {
-    Optional<WsProjectRef> project = newOptionalWsProjectRef(request.getProjectId(), request.getProjectKey());
-    boolean hasProject = project.isPresent();
-    boolean hasProjectUuid = hasProject && project.get().uuid() != null;
-    boolean hasProjectKey = hasProject && project.get().key() != null;
-
-    if (hasProjectUuid) {
-      checkProjectAdminUserByComponentUuid(userSession, project.get().uuid());
-    } else if (hasProjectKey) {
-      checkProjectAdminUserByComponentKey(userSession, project.get().key());
+  private void checkAuthorized(DbSession dbSession, SearchProjectPermissionsWsRequest request) {
+    com.google.common.base.Optional<ProjectWsRef> projectRef = newOptionalWsProjectRef(request.getProjectId(), request.getProjectKey());
+    Optional<ProjectId> projectId;
+    if (projectRef.isPresent()) {
+      projectId = Optional.of(wsSupport.findProject(dbSession, projectRef.get()));
     } else {
-      checkGlobalAdminUser(userSession);
+      projectId = Optional.empty();
     }
+    checkProjectAdmin(userSession, projectId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "checkProjectAdmin"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)